src/python/pants/engine/target.py

```diff
@@ -1062,20 +1050,15 @@ class ZipSafe(BoolField):
             default = True
     """
 
-    value: Optional[bool]
-    default: ClassVar[Optional[bool]] = None
+    expected_type = bool
+    expected_type_description = "a boolean"
 
-    @classmethod
-    def compute_value(cls, raw_value: Optional[bool], *, address: Address) -> Optional[bool]:
-        value_or_default = super().compute_value(raw_value, address=address)
-        if value_or_default is not None and not isinstance(value_or_default, bool):
-            raise InvalidFieldTypeException(
-                address,
-                cls.alias,
-                raw_value,
-                expected_type="a boolean",
-            )
-        return value_or_default
+
+class BoolField(TriBoolField, metaclass=ABCMeta):
+    """A specialization of TriBoolField which has no None state."""
```

Comment:
How about this?

```
A field that can be either True or False.

You must set the class property default or set `required = True`. See `TriBoolField` to also allow `None`.

Reply:
Note the "You must set the class property `default` or set `required = True`."

I think that we need to add some code to `compute_value()` to validate the value is never `None`. It's not enough to subclass `TriBoolField`. It would be great to add a small test to `target_test.py` that tries explicitly setting `my_bool_field=None` and confirming it errors.